packaging/src/test/java/org/alfresco/ThirdPartyLicensesIT.java

```diff
@@ -0,0 +1,114 @@
+import java.util.zip.ZipFile;
+
+import com.google.common.collect.Sets;
+
+import org.junit.Test;
+
+/**
+ * Integration tests of the declared third party licenses.
+ */
+public class ThirdPartyLicensesIT
+{
+    /** Libraries produced by Alfresco will start with "alfresco-". */
+    private static final String ALFRESCO_PREFIX = "alfresco-";
+    /** Spring Surf is also produced by Alfresco (it lives at https://github.com/Alfresco/surf). */
+    private static final String SPRING_SURF_PREFIX = "spring-surf-";
+    /** Start of the name of the zip file. */
+    private static final String ALFRESCO_SEARCH_SERVICES = "alfresco-search-services";
+
+    /**
+     * Test that the dependencies in notice.txt match the actual dependencies, to ensure we've included third party
+     * licenses for all dependencies.
+     *
+     * @throws Exception Unexpected
+     */
+    @Test
+    public void testLicensesDeclared() throws Exception
+    {
+        // Get the path to the target directory.
+        Path targetPath = Paths.get(getClass().getProtectionDomain().getCodeSource().getLocation().getPath(), "..");
+
+        // Try to find the zip file in the target directory.
+        Path zipPath = Files.find(targetPath, 1,
+                    (path, attribute) -> path.getFileName().toString().startsWith(ALFRESCO_SEARCH_SERVICES + "-")
+                                && path.toString().endsWith(".zip"))
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("Could not find " + ALFRESCO_SEARCH_SERVICES
+                                + "-*.zip in target directory. Is test being run from maven?"));
+        // Look through the zip manifest and find all the third party jar files listed.
+        Set<String> jars;
+        try (java.util.zip.ZipFile zipFile = new ZipFile(zipPath.toFile()))
+        {
+            jars = zipFile.stream()
+                        .map(ZipEntry::getName)
+                        .map(Paths::get)
+                        .map(Path::getFileName)
+                        .map(Path::toString)
+                        .filter(name -> name.endsWith(".jar"))
+                        .filter(name -> !name.startsWith(ALFRESCO_PREFIX)) // We don't need a declaration for alfresco libraries.
+                        .filter(name -> !name.startsWith(SPRING_SURF_PREFIX))
+                        .collect(toSet());
+        }
+
+        // Get the dependencies referenced in notice.txt.
+        Path noticePath = Paths.get(targetPath.toString(), "classes", "licenses", "notice.txt");
+        List<String> lines = Files.readAllLines(noticePath);
+        // Skip the header, which is all lines before the first "=== License Type ===" line.
+        int headerSize = 0;
+        for (String line : lines)
+        {
+            if (line.startsWith("==="))
+            {
+                break;
+            }
+            headerSize++;
+        }
+        Set<String> declared = lines.stream()
```

Comment:
Why don't you filter directly the stream? Something like:

`lines().stream().filter(line -> !line.startsWith("==="))...`

Reply:
I couldn't find a neat way to ignore entries from the stream before the first "===" line.  The notice.txt file has the following format:
```
Some introduction
to be ignored by test
=== Title of license ===
List
of
jars
```

Reply:
Ahhh ok, I got you, forget my comment